### PR TITLE
Add support for DLPack

### LINKED ISSUE #56: Support the new DLPack exchange protocol from Python Array API
Parallel PRs: 
- NumPy: https://github.com/numpy/numpy/pull/19083
- CuPy: https://github.com/cupy/cupy/pull/5306
- PyTorch: https://github.com/pytorch/pytorch/pull/57110

Spec: https://data-apis.org/array-api/latest/design_topics/data_interchange.html

### PR DESCRIPTION
**Pull request by dalcinl**
Fixes #56

### CODE REVIEW

**Comment by leofang** [comment]
Let me run GPU tests later today.

File: CHANGES.rst [hunk 1]
```diff
@@ -33,7 +34,10 @@ Release 3.1.0 [2021-XX-XX]
 
   + Support for weak references.
 
-.. _link: http://numba.pydata.org/numba-doc/latest/cuda/cuda_array_interface.html
+.. _DLPack: https://github.com/dmlc/dlpack
+.. _DIM: https://data-apis.org/array-api/latest/design_topics/data_interchange.html
+.. _CAI: http://numba.pydata.org/numba-doc/latest/cuda/cuda_array_interface.html
```
Comment (leofang) [inline CHANGES.rst:39]: btw they changed to RTD a while ago:
```suggestion
.. _CAI: https://numba.readthedocs.io/en/stable/cuda/cuda_array_interface.html
```

**Comment by dalcinl** [comment]
@leofang Are we good to go with this one?

**Comment by leofang** [comment]
@dalcinl I made some changes to the doc in c63cfa4. Could you take a look and then rebase/merge to resolve conflicts?

**Comment by dalcinl** [comment]
> Could you take a look and then rebase/merge to resolve conflicts?

Done. I also introduced some minor changes to your commit. The most important one is to replace CUDA -> GPU, given than the new DLPack support is more general than just NVIDIA/CUDA. Do you agree with these changes?

File: docs/source/usrman/overview.rst [hunk 2]
```diff
@@ -226,34 +226,35 @@ the predefined (i.e., :const:`MPI.SUM`, :const:`MPI.PROD`,
 :const:`MPI.MAX`, etc.)  reduction operations can be applied.
 
 
-Support for CUDA-aware MPI
+Support for GPU-aware MPI
 --------------------------
```
Comment (leofang) [inline docs/source/usrman/overview.rst:230]:
```suggestion
Support for GPU-aware MPI
-------------------------
```

**Review by leofang — APPROVED** [review]
LGTM except for one nitpick!

File: docs/source/usrman/overview.rst (hunk 2 above)
Comment (dalcinl) [reply docs/source/usrman/overview.rst:230]: Damn. Good catch!

**Review by leofang — APPROVED** [review]

File: docs/source/usrman/overview.rst [hunk 3]
```diff
@@ -226,34 +226,35 @@ the predefined (i.e., :const:`MPI.SUM`, :const:`MPI.PROD`,
...
+example, a CuPy array can be passed to a Numba CUDA-jit kernel.
+
+.. _DLPack: https://data-apis.org/array-api/latest/design_topics/data_interchange.html
+.. _CUDA Array Interface: https://numba.readthedocs.io/en/stable/cuda/cuda_array_interface.html
+
+*MPI for Python* provides an experimental support for GPU-aware MPI.
 This feature requires:
 
-1. mpi4py is built against a CUDA-aware MPI library.
+1. mpi4py is built against a CPU-aware MPI library.
```
Comment (leofang) [inline docs/source/usrman/overview.rst:247]:
```suggestion
1. mpi4py is built against a GPU-aware MPI library.
```

**Review by leofang — DISMISSED** [review]

**Review by leofang — APPROVED** [review]

**Comment by leofang** [comment]
🎉🎉🎉

**Pull request merged**